components/function-controller/design/glossary_proposal.md

```diff
@@ -0,0 +1,105 @@
+# Serverless naming convention
+
+## Problem overview
+
+Currently, Serverless in Kyma consists of two projects:
+
+- [function-controller](https://github.com/kyma-project/kyma/tree/main/components/function-controller) -
+  responsible for running a Function on a k8s cluster
+- [serverless-manager](https://github.com/kyma-project/serverless-manager) - responsible for installation and
+  configuration of Serverless
+
+Additionally, we have 3rd-party additional components like [KEDA](https://keda.sh/)
+
+In Serverless, we overuse the word "controller" which causes confusion and requires clarification. Saying "controller", we refer to:
+
+- Serverless reconcile loop
+- Serverless Pod with the reconcile loop
+- the Function Controller component in the `kyma` directory
+
+## Goal
+
+The goal of this document is to clarify the naming convention in Serverless and define its elements to avoid confusion and make it more logical.
+
+## Proposal
+
+The proposed naming conventions refer to different architecture layers of the whole project. See the [architecture](./assets/kubebuilder-architecture.png) diagram for details.
+
+### Project naming convention
+
+This section refers to the high-level architecture elements, namely to the main projects:
+
+- Serverless - the new naming convention for the `function-controller`. Serverless is responsible for running a Function on a k8s cluster. It can contain its own
+  CRD.
+- Serverless-operator - the new naming convention for `serverless-manager`. Serverless-operator installs and configures Serverless.
+- Kyma-Keda-operator - the operator which installs and configures [KEDA](https://keda.sh/).
+
+### Component naming convention
+
+This section refers to the Serverless components:
+
+- Controller - responsible for creating and configuring k8s resources to finally run a function on a cluster. It is responsible for the reconciliation of the Function CR.
+- Webhook - responsible for defaulting, validation, and conversion of the Function CR, mutating the external registry Secret, and reconciling certificates.
+
+Proposed naming convention:
+
+Deployment with the controller as the main responsibility:
+
+- ${component_name}-controller
+
+In the case of introducing a separate CRD and a separate deployment:
+
+- ${component_name}-{crd_name}-controller
+
+Deployment with webhook as main responsibility:
+
+- ${component_name}-webhook
+
+Deployment with both of those components:
+
+- ${component_name}
+
+I decided to go with the pure component name as it contains two responsibilities and from a technical perspective it's very similar to the component.
```

Comment:
```suggestion
> **NOTE:** I decided to go with the pure component name as it contains both the controller and webhook responsibilities and from the technical perspective it's very similar to the component itself. It might be confusing, and I am open to other proposals.
```